diff --git a/mupen64plus-core/src/device/r4300/r4300_core.h b/mupen64plus-core/src/device/r4300/r4300_core.h
index 5b5b7ea..30f2bbe 100644
--- a/mupen64plus-core/src/device/r4300/r4300_core.h
+++ b/mupen64plus-core/src/device/r4300/r4300_core.h
@@ -181,7 +181,7 @@ struct r4300_core
     /* FIXME: better put that near linkage_arm code
      * to help generate call beyond the +/-32MB range.
      */
-    ALIGN(4096, char extra_memory[33554432]);
+    ALIGN(16384, char extra_memory[33554432]);
     struct new_dynarec_hot_state new_dynarec_hot_state;
 #endif /* NEW_DYNAREC */
 
